src/components/new-safe/steps/Step4/useSafeCreation.ts

```diff
@@ -50,6 +54,7 @@ export const useSafeCreation = (
     if (!pendingSafe || !provider || !chain || !wallet || isCreating) return
 
     setIsCreating(true)
+    dispatch(closeAllNotifications())
```

Comment:
These notifications share the same `groupKey` and will close each other if they "overflow". I don't think we should programmatically close them like this.

Reply:
Unfortunately, we only show notifications for errors. This adds a bit of confusion imo when retrying the tx and the error notification is still being displayed. In the past we looked into duplicating the `TxEvents` flows for safe creation i.e. showing notifications for signing etc. but decided against it as we wanted to use inline messages. Might be worth reconsidering but I don't want to blow up this PR.

Reply:
That makes sense. I would adjust `closeAllNotifications` to `closeByGroupKey` to prevent the closing of unrelated notifcations though.